tcblib/db_functions.py

```diff
@@ -0,0 +1,97 @@
+import boto3
+import datetime
+import os
+
+DB = boto3.client('dynamodb')
+TABLE = os.environ['DYNAMODB_TABLE']
+DEFAULT_HOST_ID = os.environ['DEFAULT_HOST_ID']
+DEFAULT_HOST_NAME = os.environ['DEFAULT_HOST_NAME']
+
+def get_date():
+    """Get date for next tech chat (will return today if today is Friday)."""
+    today = datetime.date.today()
+    friday = today + datetime.timedelta((4-today.weekday()) % 7)
+    return friday.strftime('%Y%m%d')
+
+
+def query_host(user_attr):
+    """Query for the host, for either the name or user_id."""
+    response = DB.get_item(
+        TableName=TABLE,
+        Key={
+            'Date': {'N': '{0}'.format(get_date())}
+        },
+        ProjectionExpression='Host.#attr',
+        ExpressionAttributeNames={
+            '#attr': user_attr
+        }
+    )
+
+    if 'Item' not in response:
+        update_host(DEFAULT_HOST_ID, DEFAULT_HOST_NAME)
```

Comment:
If you have a default, it's a little redundant to update something to the default. Think about treating the table as a "list of overrides", because lets face it: you're probably the default emcee for a while

Reply:
in this case, 'update_host' is actually *adding* the host -- like, this only happens when someone tries to query to find out who the host is, and there isn't one yet. it's 'update_or_add_if_not_exists_host', technically.